### PR TITLE
Feature/pelagos 4604 show server errors for submission

### PR DESCRIPTION
**Pull request by mickel1138**
Different approach from PR #993

### CODE REVIEW

File: assets/static/js/dataset-submission.js [hunk 1]
```diff
@@ -256,6 +279,7 @@ $(function() {
             url: url + "/" + datasetSubmissionId + "?validate=false",
             method: "PUT",
             data: formData,
+            timeout: 30000,
```
Comment (fightingtexasaggie) [inline assets/static/js/dataset-submission.js:282]: For these PUT operations, I think 30 seconds is really generous. I'd think what doesn't finish in 5 seconds will likely not finish in 30.  Maybe we should shorten?

**Review by fightingtexasaggie — COMMENTED** [review]
currently the auto-save when switching between tabs is no longer working, and also the hidden divs with the error messages are being exposed...see screenshot.

Screenshot from 2021-09-13 15-26-22

File: assets/static/js/dataset-submission.js (hunk 1 above)
Comment (mickel1138) [reply assets/static/js/dataset-submission.js:282]: It's less than the 60 seconds before, but less than 30 would be better, not sure what the magic number would be.

**Comment by mickel1138** [comment]
> currently the auto-save when switching between tabs is no longer working, and also the hidden divs with the error messages are being exposed...see screenshot.
> 
Good catch, I will get rid of it. The auto save should work, but i'll check the logic.

**Comment by mickel1138** [comment]
> > currently the auto-save when switching between tabs is no longer working, and also the hidden divs with the error messages are being exposed...see screenshot.
> 
> Good catch, I will get rid of it. The auto save should work, but i'll check the logic.

Ok, I double checked, and the DIVs don't show, this may be a integrity check/cache issue.

**Review by fightingtexasaggie — DISMISSED** [review]
This works. Maybe a later improvement might be to display a "saving your information..." along with a "done" noty where they close maybe a second after success. If  the server is down, the user will still see the attempt is being made, with this error dialog after 30 sec.

**Review by praneethpr — APPROVED** [review]
Tested and works

File: assets/static/js/dataset-submission.js [hunk 2]
```diff
@@ -279,22 +303,22 @@ $(function() {
...
-                  var n = noty(
-                  {
-                      layout: "top",
-                      theme: "relax",
-                      type: "error",
-                      text: message,
-                      modal: true,
-                  });
+                } else {
+                    if (jqXHR.status != 0) {
```
Comment (praneethpr) [inline assets/static/js/dataset-submission.js:310]: Add type comparison

**Review by praneethpr — APPROVED** [review]

**Review by fightingtexasaggie — APPROVED** [review]

**Pull request merged**